models/repo.go

```diff
@@ -2099,7 +2099,7 @@ func ForkRepository(u *User, oldRepo *Repository, name, desc string) (_ *Reposit
 		repoPath, fmt.Sprintf("ForkRepository(git update-server-info): %s", repoPath),
 		"git", "update-server-info")
 	if err != nil {
-		return nil, fmt.Errorf("git update-server-info: %v", err)
+		return nil, fmt.Errorf("git update-server-info: %v", stderr)
```

Comment:
Figured, it's just process.ExecDir in modules/process/manager.go that would need a somewhat extended documentation. All returns are in the form:

- return "", err.Error(), err
- return "", ErrExecTimeout.Error(), ErrExecTimeout
- return bufOut.String(), bufErr.String(), err

where the `err` variable in the last case is a return from cmd.Wait (exec.ExitError) while bufErr.String() is the stderr.

So the *signature* of ExecDir has a duplicated return value, because the "stderr" part is always included in the "err".
I suggest ExecDir signature is changed, to reduce confusion.